### PR TITLE
Use MeshModel interface for conversion of application to design-file

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
**Notes for Reviewers**
Use MeshModel interface for converting Applications to design-file.
Problems with OAM:
1. Not compatible in the current UI in meshery-extensions.
2. The metadata that is required to style and get information of each entity is missing as well in oam implementation.

Sample request:
```
{
  "application_data": {
    "name": "cluster-role.yml",
    "application_file": "apiVersion: rbac.authorization.k8s.io/v1\nkind: Role\nmetadata:\n  namespace: default\n  name: pod-reader\nrules:\n- apiGroups: [\"\"] # \"\" indicates the core API group\n  resources: [\"pods\"]\n  verbs: [\"get\", \"watch\", \"list\"]\n"
  },
  "save": false
}
```
response:
```
[
    {
        "name": "cluster-role.yml",
        "application_file": "name: Autogenerated\nservices:\n  d29b8b29-1ea7-4b28-ae3e-3228e49eead8:\n    name: pod-reader\n    type: Role\n    apiVersion: rbac.authorization.k8s.io/v1\n    namespace: default\n    model: kubernetes\n    settings:\n      rules:\n      - api Groups:\n        - \"\"\n        resources:\n        - pods\n        verbs:\n        - get\n        - watch\n        - list\n    traits:\n      meshmap:\n        meshmodel-metadata:\n          genealogy: \"\"\n          isCustomResource: false\n          isNamespaced: true\n          logoURL: https://github.com/cncf/artwork/blob/master/projects/kubernetes/icon/white/kubernetes-icon-white.svg\n          model: kubernetes\n          modelDisplayName: Kubernetes\n          primaryColor: '#326CE5'\n          published: true\n          secondaryColor: '#7aa1f0'\n          shape: round-rectangle\n          styleOverrides: '{\"height\":\"22px\",\"width\":\"16px\",\"padding\":\"6px\"}'\n          subCategory: Scheduling & Orchestration\n          svgColor: ui/public/static/img/meshmodels/kubernetes/color/clusterrole-color.svg\n          svgComplete: \"\"\n          svgWhite: ui/public/static/img/meshmodels/kubernetes/white/clusterrole-white.svg\n",
        "user_id": null,
        "location": {
            "branch": "",
            "host": "",
            "path": "",
            "type": "local"
        },
        "type": {
            "String": "Kubernetes Manifest",
            "Valid": true
        },
        "source_content": "YXBpVmVyc2lvbjogcmJhYy5hdXRob3JpemF0aW9uLms4cy5pby92MQpraW5kOiBSb2xlCm1ldGFkYXRhOgogIG5hbWVzcGFjZTogZGVmYXVsdAogIG5hbWU6IHBvZC1yZWFkZXIKcnVsZXM6Ci0gYXBpR3JvdXBzOiBbIiJdICMgIiIgaW5kaWNhdGVzIHRoZSBjb3JlIEFQSSBncm91cAogIHJlc291cmNlczogWyJwb2RzIl0KICB2ZXJiczogWyJnZXQiLCAid2F0Y2giLCAibGlzdCJdCg=="
    }
]
```

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/pattern/core/pattern.go [hunk 1]
```diff
@@ -532,12 +534,32 @@ func createPatternServiceFromK8s(manifest map[string]interface{}) (string, Servi
...
 
-	if len(w) == 0 {
+	// Get MeshModel entity with the selectors
+	res := regManager.GetEntities(&meshmodelv1alpha1.ComponentFilter{
+		Name:       kind,
+		ModelName:  "kubernetes", // currently hardcoded for kubernetes manifest conversion
+		APIVersion: apiVersion,
+		Greedy:     false,
+	})
+	if len(res) == 0 {
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply server/models/pattern/core/pattern.go:545]: @Abhishek-kumar09 will you open a separate issue for this?

**Review by leecalcote — DISMISSED** [review]

File: server/models/pattern/core/pattern.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Abhishek-kumar09) [reply server/models/pattern/core/pattern.go:545]: `res` is the array of meshmodel components that matches the selected filter sent. len=0 denotes unregistered component or bad filter. So

[the remark this answers is not in the document]
Comment (leecalcote) [reply server/models/pattern/core/pattern.go:545]: @Abhishek-kumar09 thanks for this explanation. Please paste this into a code comment.

Please offer a more descriptive variable name.

To my point, however, I'm looking for a more robust conditional to be used. Is the nil case accounted for? the blank (whitespace) case accounted for?

[the remark this answers is not in the document]
Comment (Abhishek-kumar09) [reply server/models/pattern/core/pattern.go:545]: okay, I could do it.

File: server/models/pattern/core/pattern.go [hunk 2]
```diff
@@ -532,12 +534,32 @@ func createPatternServiceFromK8s(manifest map[string]interface{}) (string, Servi
 	if apiVersion == "" || kind == "" {
 		return "", Service{}, ErrCreatePatternService(fmt.Errorf("empty apiVersion or kind in manifest"))
 	}
-	w := GetWorkloadsByK8sAPIVersionKind(apiVersion, kind)
 
-	if len(w) == 0 {
+	// Get MeshModel entity with the selectors
+	res := regManager.GetEntities(&meshmodelv1alpha1.ComponentFilter{
+		Name:       kind,
+		ModelName:  "kubernetes", // currently hardcoded for kubernetes manifest conversion
```
Comment (Revolyssup) [inline server/models/pattern/core/pattern.go:541]: Please omit `modelname` and greedy is by default false.

File: server/models/pattern/core/pattern.go [hunk 3]
```diff
@@ -558,13 +580,14 @@ func createPatternServiceFromK8s(manifest map[string]interface{}) (string, Servi
 	rest = Format.Prettify(rest, false)
 	svc := Service{
 		Name:        name,
-		Type:        w[0].OAMDefinition.Name,
+		Type:        kind,
 		APIVersion:  apiVersion,
 		Namespace:   namespace,
 		Model:       "kubernetes",
```
Comment (Revolyssup) [inline server/models/pattern/core/pattern.go:586]: use the modelname retrieved from Component definition.

File: server/models/pattern/core/pattern.go [hunk 4]
```diff
@@ -558,13 +580,14 @@ func createPatternServiceFromK8s(manifest map[string]interface{}) (string, Servi
 	rest = Format.Prettify(rest, false)
 	svc := Service{
 		Name:        name,
-		Type:        w[0].OAMDefinition.Name,
+		Type:        kind,
 		APIVersion:  apiVersion,
```
Comment (Revolyssup) [inline server/models/pattern/core/pattern.go:584]: add comp.APIVersion

File: server/models/pattern/core/pattern.go [hunk 5]
```diff
@@ -558,13 +580,14 @@ func createPatternServiceFromK8s(manifest map[string]interface{}) (string, Servi
 	rest = Format.Prettify(rest, false)
 	svc := Service{
 		Name:        name,
-		Type:        w[0].OAMDefinition.Name,
+		Type:        kind,
```
Comment (Revolyssup) [inline server/models/pattern/core/pattern.go:583]: Add comp.Name

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: server/models/pattern/core/pattern.go (hunk 2 above)
Comment (Abhishek-kumar09) [reply server/models/pattern/core/pattern.go]: Done

File: server/models/pattern/core/pattern.go (hunk 3 above)
Comment (Abhishek-kumar09) [reply server/models/pattern/core/pattern.go]: Done

**Review by Revolyssup — APPROVED** [review]

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**